src/node_api_jsrt.cc

```
@@ -176,6 +217,9 @@ namespace jsrtimpl {
       jsrtimpl::ExternalCallback* externalCallback =
         reinterpret_cast<jsrtimpl::ExternalCallback*>(callbackState);
 
+      // Make sure any errors encountered last time we were in N-API are gone.
```

Is this the behavior in v8 too? I'm not sure if we want the previous error to be silently swallowed here

---

Yes, this is correct, and is consistent with NAPI-V8. The "last error" should only apply to the _last_ call from JavaScript into C++, so any error from a prior call should be cleared.